src/graphforge/executor/evaluator.py

```
@@ -189,6 +203,84 @@ def evaluate_expression(expr: Any, ctx: ExecutionContext) -> CypherValue:
+                        else str(right_val.value)
+                    )
+                    return CypherString(left_str + right_str)
+
+            # Type checking: both operands must be numeric
+            if not isinstance(left_val, (CypherInt, CypherFloat)) or not isinstance(
+                right_val, (CypherInt, CypherFloat)
+            ):
+                raise TypeError(
+                    f"Arithmetic operator {expr.op} requires numeric operands, "
+                    f"got {type(left_val).__name__} and {type(right_val).__name__}"
+                )
```

Ruff (0.14.14)

[warning] Avoid specifying long messages outside the exception class

(TRY003)